examples/topics/categorical/correlograms.py

```diff
@@ -0,0 +1,68 @@
+''' A categorical plot showing the correlations in mineral content for 214 samples of glass fragments
+obtained during forensic work. The dataset contains seven variables measuring the amounts of magnesium (Mg),
+calcium (Ca), iron (Fe), potassium (K), sodium (Na), aluminum (Al), and barium (Ba) found in each glass fragment.
+The colored circles represent the correlations between pairs of these variables.
+The magnitude of each correlation is encoded in the size of the circles.
+
+.. bokeh-example-metadata::
+    :sampledata: forensic_glass
+    :apis: bokeh.plotting.figure.scatter
+    :refs: :ref:`ug_topics_categorical_scatters`
+    :keywords: scatter, correlogram
+
+'''
+import numpy as np
+import pandas as pd
+
+from bokeh.models import ColorBar, ColumnDataSource, FixedTicker, LinearColorMapper
+from bokeh.plotting import figure, show
+from bokeh.sampledata.forensic_glass import data
+from bokeh.transform import transform
+
+df = data.copy()
+
+pairs = ["Mg-Ba", "Mg-Al", "Mg-Na", "Mg-K", "Mg-Fe", "Mg-Ca", "Ca-Ba",
+         "Ca-Al", "Ca-Na", "Ca-K", "Ca-Fe", "Fe-Ba", "Fe-Al", "Fe-Na",
+         "Fe-K", "K-Ba", "K-Al", "K-Na", "Na-Ba", "Na-Al", "Al-Ba"]
```

Comment:
I'd encourage you to explore an approach that starts with just a list of the elements, and leverages `itertools` to make the combinations. This would probably allow some of the code below for the "splitting" to be simplified or eliminated
```python
In [17]: from itertools import combinations

In [18]: elts = ("Mg", "Ca", "Fe", "K", "Na", "Al")

In [19]: list(combinations(elts, 2))
Out[19]:
[('Mg', 'Ca'),
 ('Mg', 'Fe'),
 ('Mg', 'K'),
 ('Mg', 'Na'),
 ('Mg', 'Al'),
 ('Ca', 'Fe'),
 ('Ca', 'K'),
 ('Ca', 'Na'),
 ('Ca', 'Al'),
 ('Fe', 'K'),
 ('Fe', 'Na'),
 ('Fe', 'Al'),
 ('K', 'Na'),
 ('K', 'Al'),
 ('Na', 'Al')]
```

Reply:
Here's a couple of ideas to play around with that could probably simplify a few things:
```python
In [28]: pairs = list(combinations(elts, 2))

In [29]: x, y = list(zip(*pairs))

In [31]: for x, y in pairs:
    ...:     print(x, y)
```

Reply:
I tried using `combinations` to create the pairs but the plot appeared in reverse order than intended, so I used nested loops to create the pairs in reverse order instead. It seems `combinations` pairs the elements in order.

Here's what I mean. Using `itertools.combinations`:

<img width="624" alt="Screenshot 2023-10-04 at 12 12 09 PM" src="https://github.com/bokeh/bokeh/assets/99359668/bfa855db-2a86-4e14-8b2c-64d22a8ab16b">

Using nested loops:

<img width="602" alt="Screenshot 2023-10-04 at 12 12 20 PM" src="https://github.com/bokeh/bokeh/assets/99359668/23dd3283-56a5-4bfe-bc9c-cbe4b46ddac9">

Reply:
The order on the axis is controlled by the order of factors passed to `y_range` I think the only change necessary is to call `reversed` on the list of factors you pass for the range.

Reply:
Thanks!